### ISSUE TITLE
[Bug]: cant change brightness to a Z2M group of lights

### ISSUE THREAD

**Issue by stgl83** [t0bst4r/home-assistant-matter-hub#336]
### Version

3.0.0-alpha.56

### Matter Controller

Alexa

### Steps to reproduce

I created a group auf lights in Z2M which all have the possibility to configure brightness and color temperature. If i try to change the brigthness of the group entity via Alexa and HAMB it won't work. Switching the lights on/off is working.

If i change brightness of the group members directly everything works.

### Relevant log output

```shell
[ 2024-12-20T07:56:37.051Z ] [ ERROR ] [ apply-patch-state                                  ]: Failed to patch the following properties:
{
  "startUpColorTemperatureMireds": 454.1326067211626,
  "colorTemperatureMireds": 454.1326067211626
}
[ 2024-12-20T07:56:37.055Z ] [ ERROR ] [ Matter / Transaction                               ]: Error post-commit of 2273e952a2df436fbb9c7fa8d3abf968.aggregator.light_wohnzimmer_licht.homeAssistantEntity.state: Error in reactor<2273e952a2df436fbb9c7fa8d3abf968.aggregator.light_wohnzimmer_licht.colorControl.update>: Validating 2273e952a2df436fbb9c7fa8d3abf968.aggregator.light_wohnzimmer_licht.colorControl.state: Constraint "colorTempPhysicalMinMireds to colorTempPhysicalMaxMireds": Value 454.1326067211626 is not within bounds defined by constraint (135)
  at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/validation/constraint.js:35:17
  at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/validation/ValueValidator.js:155:25
  at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/validation/conformance.js:15:22
  at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/managed/values/StructManager.js:135:13
  at Object.change (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/managed/Datasource.js:172:7)
  at ColorControl$State.set [as colorTemperatureMireds] (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/managed/values/StructManager.js:105:32)
  at Function.assign (<anonymous>)
  at applyPatchState (file:///usr/local/lib/node_modules/home-assistant-matter-hub/dist/backend/cli.js:1055:14)
  at ColorControlServer.update (file:///usr/local/lib/node_modules/home-assistant-matter-hub/dist/backend/cli.js:1155:5)
  at #reactWithLocks (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/internal/Reactors.js:286:26)
[ 2024-12-20T07:56:38.615Z ] [ INFO  ] [ Matter / InteractionServer                         ]: Successfully created subscription 1016204450 for Session 57726. Updates: 1 - 300 => 186 seconds (sendInterval = 93 seconds)
[ 2024-12-20T07:56:48.453Z ] [ INFO  ] [ Matter / Behavior                                  ]: Invoke *2273e952a2df436fbb9c7fa8d3abf968.aggregator.light_wohnzimmer_licht.onOff.on* online#653f187@2a8a4918a1ac661 (no payload)
```

### Documentation & Issues

- [X] I have reviewed the [documentation](https://github.com/t0bst4r/home-assistant-matter-hub/blob/main/packages/documentation/README.md) and the linked troubleshooting guide.
- [X] I have searched the [issues](https://github.com/t0bst4r/home-assistant-matter-hub/issues) for a similar problem.

**Comment by kennylevinsen** [comment]
Can you share the state for your Z2M light group from Home assistant? Go to Developer Tools → States and find the light entity for your group.

**Comment by stgl83** [comment]
min_color_temp_kelvin: 2702
max_color_temp_kelvin: 4000
min_mireds: 250
max_mireds: 370
effect_list:
  - blink
  - breathe
  - okay
  - channel_change
  - finish_effect
  - stop_effect
supported_color_modes:
  - color_temp
effect: null
color_mode: null
brightness: null
color_temp_kelvin: null
color_temp: null
hs_color: null
rgb_color: null
xy_color: null
friendly_name: Wohnzimmer Licht
supported_features: 44

I found something strange. Some states have a missmatch to the group

Lamp 1 & 2:
min_color_temp_kelvin: 2702
max_color_temp_kelvin: 6535
min_mireds: 153
max_mireds: 37

Lamp 3:
min_color_temp_kelvin: 2202
max_color_temp_kelvin: 4000
min_mireds: 250
max_mireds: 454

**Comment by kennylevinsen** [comment]
Yeah so the issue is that your lights have mixed capabilities with 1 & 2 supporting a warmer temperature range (153 to 370) while lamp 3 supports a colder range (250 to 454). The reported min/max for the group being the intersection of capabilities.

If lamp 3 is then on at its coldest setting, you'll probably get a report where the light group has a color temperature that's entirely outside the reported supported range of the group. When home-assistant-matter-hub updates the state on matter's side of things, matter.js rejects the invalid state.

A hack on our side would be to update min/max if the current value is outside the range.

**Comment by t0bst4r** [comment]
hmm. we can add it as a quick fix, but somehow that solution seems wrong. lets give it a try.

**Comment by kennylevinsen** [comment]
> Hmm. we can add it as a quick fix, but somehow that solution seems wrong.

Yeah it's just a workaround. When what we get from HA/Z2M is nonsense that wouldn't be protocol compliant I'm not sure we have much in the way of pretty solutions. :/

Alternatives I can think of:
1. Get Z2M (or HA, depending on whether it is a Z2M group or HA group) to return the union of available colors rather than intersection as min/max and rely on it capping colors to individual device limits when setting the group.
2. Have us cap values to the reported range, which in turn means that matter controllers will not know the actual current color when it is out of range.

**Comment by stgl83** [comment]
Hi t0bst4r,

thanks for the quick fix.
Now i get an error while starting the bridge :(

[ 2024-12-20T19:23:38.157Z ] [ ERROR ] [ Bridge / 2273e952a2df436fbb9c7fa8d3abf968          ]: Failed to start bridge due to error: Behaviors have errors
  at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/endpoint/properties/Behaviors.js:128:21
  at all (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/general/dist/esm/util/Construction.js:375:17)
  at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/general/dist/esm/util/Construction.js:365:11
  Cause #0: Error initializing 2273e952a2df436fbb9c7fa8d3abf968.aggregator.light_wohnzimmer_licht.colorControl
    at crash (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/internal/BehaviorBacking.js:48:13)
  Caused by: Validating 2273e952a2df436fbb9c7fa8d3abf968.aggregator.light_wohnzimmer_licht.colorControl.state: Constraint "colorTempPhysicalMinMireds to colorTempPhysicalMaxMireds": Value 454.1326067211626 is not within bounds defined by constraint (135)
    at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/validation/constraint.js:35:17
    at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/validation/ValueValidator.js:155:25
    at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/validation/conformance.js:15:22
    at file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/managed/values/StructManager.js:135:13
    at Object.change (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/managed/Datasource.js:172:7)
    at ColorControl$State.set [as colorTemperatureMireds] (file:///usr/local/lib/node_modules/home-assistant-matter-hub/node_modules/@matter/node/dist/esm/behavior/state/managed/values/StructManager.js:105:32)
    at Function.assign (<anonymous>)
    at applyPatchState (file:///usr/local/lib/node_modules/home-assistant-matter-hub/dist/backend/cli.js:1055:14)
    at ColorControlServer.update (file:///usr/local/lib/node_modules/home-assistant-matter-hub/dist/backend/cli.js:1155:5)
    at ColorControlServer.initialize (file:///usr/local/lib/node_modules/home-assistant-matter-hub/dist/backend/cli.js:1147:10)

**Comment by t0bst4r** [comment]
I just noticed I only fixed it for level control, but not for color temperature. 😅 so it's still the same error, but now it's failing during initialization because your lights are already on. Turn the lights off, or set a temperature which would be allowed, and start the addon. It will boot, but after that you'll get the same error when changing the temperature to an out-of-range value.

Unfortunately we just left home for Christmas so I won't be able to fix it before 27th.

@ Whoever reads this, feel free to create PR. I'll review it from my mobile as soon as possible.
